src/passes/function_passes/CheckPointPass.cpp

```diff
@@ -0,0 +1,157 @@
+#include <vector>
+#include <map>
+#include <stdint.h>
+#include <llvm/Pass.h>
+#include <llvm/IR/Intrinsics.h>
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/IR/BasicBlock.h>
+#include <llvm/IR/Function.h>
+#include <llvm/IR/Module.h>
+#include <llvm/IR/DataLayout.h>
+#include <llvm/IR/Instruction.h>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/Type.h>
+#include <llvm/Support/raw_ostream.h>
+#include <llvm/Transforms/IPO/PassManagerBuilder.h>
+#define TRACE_FUN_NAME "trace"
+#define GUARD_FUN_NAME "__guard_failure"
+#define UNOPT_PREFIX "__unopt_"
+
+using namespace llvm;
+
+namespace {
+
+  bool isJump(unsigned);
```

Comment:
Perhaps use 4-space indent?

Reply:
The LLVM people use 2-space indent for their C++ code, I was just trying following their style. I can change it though